p8e-sdk/src/main/kotlin/io/p8e/ContractManager.kt

```diff
@@ -144,8 +147,10 @@ class ContractManager(
         val errorHandler: ContractErrorHandler<T>
     )
 
-    private val heartbeatConnections = ConcurrentHashMap<HeartbeatConnectionKey, HeartbeatQueuer>()
```

Comment:
This could have been the source of some of the problems. The heartbeatConnections held a reference to the queuer. This means that it's possible the queuer could have been out of sync from the queuers array and the heartbeat was pinging a queuer that wasn't present in the queuers map. This would mean it would be connected but couldn't receive or process events.